### PR TITLE
[MRG] Add a command to generate report for local git repository

### LINKED ISSUE #244: Support generating reports for local git repository
It would be great if the `report` feature can summarize local git repos.

### PR DESCRIPTION
**Pull request by zlfben**
Closes #244 

```bash
cd <project name>
mle report-local --email=<git email> --start-date=YYYY-MM-DD --end-date=YYYY-MM-DD <path_to_git_repo>
```

- `--start-date` and `--end-date` are optional parameters. If omitted, the command will generate a report for the default date range of the last 7 days.
- Replace `<git email>` with your Git email and `<path_to_git_repo>` with the path to your local Git repository.

### CODE REVIEW

File: mle/integration/local_git.py [hunk 1]
```diff
@@ -111,3 +114,48 @@ def get_commit_diff(self, commit_hash, show_content=False):
...
+        """
+        Get readme content of the repository.
+        :return: The readme content
+        """
+        content = self.get_source_code("README.md")
+        if len(content):
+            return list(content.values())[0]
+        return None
+
+    def get_structure(self, path='', include_invisible=False):
```
Comment (huangyz0918) [inline mle/integration/local_git.py:148]: there is a `get_structure` function in `util`, you can modify to reuse

Comment (zlfben) [reply mle/integration/local_git.py]: Have to directly use "git ls-files", which is different from both approach.

The issue is with git-ignored folders like `.venv`. They easily blow up openai's token limits.

**Comment by zlfben** [comment]
Tested successful with the default 7 day period. I'll need to add some more parameters like start-date and end-date.

**Review by huangyz0918 — APPROVED** [review]
LGTM.

I think maybe we should keep only one `SummaryAgent` and use different functions like `summarize_git` and `summarize_github`. What do you think?

**Pull request merged**